src/test/java/seedu/address/logic/commands/MarkTaskCommandTest.java

```diff
@@ -0,0 +1,119 @@
+
+        assertCommandSuccess(markTaskCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_invalidIndexUnfilteredList_throwsCommandException() {
+        Index outOfBoundIndex = Index.fromOneBased(model.getFilteredTaskList().size() + 1);
+        MarkTaskCommand markTaskCommand = new MarkTaskCommand(outOfBoundIndex);
+
+        assertCommandFailure(markTaskCommand, model, Messages.MESSAGE_INVALID_TASK_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void execute_markAlreadyMarkedTask_throwsCommandException() {
+        Task taskToMark = model.getFilteredTaskList().get(INDEX_FIRST_PERSON.getZeroBased());
+        model.markTask(taskToMark);
+        MarkTaskCommand markTaskCommand = new MarkTaskCommand(INDEX_FIRST_PERSON);
+
+        assertCommandFailure(markTaskCommand, model, MarkTaskCommand.MESSAGE_HAS_BEEN_MARKED);
+    }
+
+    @Test
+    public void execute_validIndexFilteredList_success() {
+        showTaskAtIndex(model, INDEX_FIRST_PERSON);
+
+        Task taskToMark = model.getFilteredTaskList().get(INDEX_FIRST_PERSON.getZeroBased());
+        MarkTaskCommand markTaskCommand = new MarkTaskCommand(INDEX_FIRST_PERSON);
+
+        Model expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
+        showTaskAtIndex(expectedModel, INDEX_FIRST_PERSON);
+        Task markedTask = expectedModel.markTask(taskToMark);
+
+        String expectedMessage = String.format(MarkTaskCommand.MESSAGE_MARK_TASK_SUCCESS,
+                Messages.format(markedTask));
+
+        assertCommandSuccess(markTaskCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_invalidIndexFilteredList_throwsCommandException() {
+        showTaskAtIndex(model, INDEX_FIRST_PERSON);
+
+        Index outOfBoundIndex = INDEX_SECOND_PERSON;
+        // ensures that outOfBoundIndex is still in bounds of address book list
+        assertTrue(outOfBoundIndex.getZeroBased() < model.getAddressBook().getTaskList().size());
+
+        MarkTaskCommand markTaskCommand = new MarkTaskCommand(outOfBoundIndex);
+
+        assertCommandFailure(markTaskCommand, model, Messages.MESSAGE_INVALID_TASK_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void equals() {
+        MarkTaskCommand markTaskFirstCommand = new MarkTaskCommand(INDEX_FIRST_PERSON);
+        MarkTaskCommand markTaskSecondCommand = new MarkTaskCommand(INDEX_SECOND_PERSON);
+
+        // same object -> returns true
+        assertTrue(markTaskFirstCommand.equals(markTaskFirstCommand));
+
+        // same values -> returns true
+        MarkTaskCommand markTaskFirstCommandCopy = new MarkTaskCommand(INDEX_FIRST_PERSON);
+        assertTrue(markTaskFirstCommand.equals(markTaskFirstCommandCopy));
+
+        // different types -> returns false
+        assertFalse(markTaskFirstCommand.equals(1));
+
+        // null -> returns false
+        assertFalse(markTaskFirstCommand.equals(null));
+
+        // different person -> returns false
```

Comment:
Maybe can update the INDEX_FIRST_PERSON here also?

```suggestion
        // different index -> returns false
```